Sources/Common/Service/UserAgentUtil.swift

```diff
@@ -23,7 +25,7 @@ public class UserAgentUtilImpl: UserAgentUtil {
      * `Customer.io iOS Client/1.0.0-alpha.16`
      */
     public func getUserAgentHeaderValue() -> String {
-        var userAgent = "Customer.io \(deviceInfo.osName ?? "iOS") Client/\(deviceInfo.sdkVersion)"
+        var userAgent = "Customer.io \(sdkClient)"
```

Comment:
I'm not fully sure what exactly `deviceInfo.osName` will return in different cases, so I tried to cover cases for all `iOS`, `visionOS`, and wrappers using the same path.

If I understand your suggestion correctly, you're proposing we use `visionOS` irrespective of `SdkClient` source if `deviceInfo.osName` returns `visionOS`. This means it won't be as flexible as Android or how it ideally should be, but we would need to take the following assumptions if we follow this approach:

1. We are making this change primarily to simplify `SdkClient` creation (I don't see any other significant benefit, but please let me know if you have something in mind)
2. Any value in `SdkClient` source will be ignored if `deviceInfo.osName` returns `visionOS`
3. Any value in `deviceInfo.osName` will be ignored if it's not `visionOS`, and `SdkClient` source will be used instead which will default to `iOS` for native SDK
4. Support for any other native platforms (similar to visionOS) won't be considered unless we explicitly add support for them. For now, it will be limited to just `visionOS`, and not any more generic implementation (like a supported OS list) to keep the scope lighter. If needed in future, we can add the support later.

Does this make sense? If so, I'm okay with making the changes but wanted to ensure we're setting the right expectations. Also, this is a heavier change and will impact all the tests I've added/fixed. So just wanted to double-check before we proceed with making the change.

Reply:
Discussed on call, added a suggetion as a commit.